src/common/ceph_context.h

```diff
@@ -261,6 +262,15 @@ class CephContext {
   void notify_pre_fork();
   void notify_post_fork();
 
+  void set_mon_addrs(const std::list<entity_addrvec_t>& in) {
+    // protect updates to list, to avoid racy updates and gets?
```

Comment:
It is possibly better to store this in cct->_conf (ConfigProxy), that has seems to have reader/writer protection in it, looking at this aspect further.

Reply:
As we discussed, keeping this in `CephContext` is more appropriate as we don't want to pay the cost of encoding/decoding the mon addrs as a config.